src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -321,9 +344,24 @@ public function commit()
                 $ok = false;
                 $type = is_object($v) ? get_class($v) : gettype($v);
                 CacheItem::log($this->logger, 'Failed to save key "{key}" ({type})', array('key' => substr($id, strlen($this->namespace)), 'type' => $type, 'exception' => $e instanceof \Exception ? $e : null));
+
+                foreach ($tags as $tag => $ids) {
+                    unset($tags[$tag][$id]);
+                }
+            }
+        }
+
+        if ($tags) {
+            try {
+                $ok = $this->doTag($tags) && $ok;
+            } catch (\Exception $e) {
+                if (!$this instanceof TagInvalidationInterface) {
+                    throw $e;
+                }
+                $ok = false;
+                CacheItem::log($this->logger, 'Failed to commit cache tags', array('exception' => $e));
             }
         }
-        $this->deferred = array();
```

Comment:
this should be sent separately as it can be applied without waiting for 3.2 then